src/components/Popups/SignUpPopup.js

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import PopupWithForm from "./PopupWithForm";
+
+function SignUpPopup({ isOpen, onClose, handleSubmit, openSignIn, message }) {
+
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [username, setUsername] = React.useState("");
+
+  function handleEmailChange(e) {
+    setEmail(e.target.value);
+  }
+
+  function handlePasswordChange(e) {
+    setPassword(e.target.value);
+  }
+
+  function handleUserNameChange(e) {
+    setUsername(e.target.value);
+  }
+
+  function onSubmit(e) {
+    e.preventDefault();
+    handleSubmit(email, password, username);
+  }
+
+
+  return (
+    <PopupWithForm
+      title="Sign Up"
+      formName="signUp"
+      isOpen={isOpen}
+      submitText="Sign Up"
+      onSubmit={onSubmit}
+      onClose={onClose}
+      altOption="Sign in"
+      handleAlt={openSignIn}
+    >
+      <label htmlFor="emailsignup" className="popup__input-label">
+        Email
+      </label>
+      <input
+        className="popup__input popup__input_info_email"
+        type="email"
+        placeholder="Enter email"
+        name="email"
+        id="emailsignup"
+        minLength="2"
+        maxLength="40"
+        onChange={handleEmailChange}
```

Comment:
* If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.